src/ui/public/vis/lib/calculate_object_hash.js

```diff
@@ -0,0 +1,60 @@
+function pad(hash, len) {
+  while (hash.length < len) {
+    hash = '0' + hash;
+  }
+  return hash;
+}
+
+function fold(hash, text) {
+  let i;
+  let chr;
+  let len;
+  if (text.length === 0) {
+    return hash;
+  }
+  for (i = 0, len = text.length; i < len; i++) {
+    chr = text.charCodeAt(i);
+    hash = ((hash << 5) - hash) + chr;
+    hash |= 0;
+  }
+  return hash < 0 ? hash * -2 : hash;
+}
+
+function foldObject(hash, o, seen) {
+  return Object.keys(o).sort().reduce(foldKey, hash);
+  function foldKey(hash, key) {
+    return foldValue(hash, o[key], key, seen);
+  }
+}
+
+function foldValue(input, value, key, seen) {
+  const hash = fold(fold(fold(input, key), toString(value)), typeof value);
+  if (value === null) {
+    return fold(hash, 'null');
+  }
+  if (value === undefined) {
+    return fold(hash, 'undefined');
+  }
+  if (typeof value === 'object') {
+    // skip vis object from hash calculation
+    if (key === 'vis') {
```

Comment:
what if this is not the vis object, but a key for a deeper row-value?

Reply:
generally the rows are an array so this should not affect anything, but just to be sure i added a comparison that this is actually a Vis type